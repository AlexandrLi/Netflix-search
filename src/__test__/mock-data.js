export const movies = [{
    vote_count: 8410,
    id: 122,
    video: false,
    vote_average: 8.2,
    title: 'The Lord of the Rings: The Return of the King',
    popularity: 22.480443,
    poster_path: '/uexxR7Kw1qYbZk0RYaF9Rx5ykbj.jpg',
    original_language: 'en',
    original_title: 'The Lord of the Rings: The Return of the King',
    genre_ids: [
      12,
      14,
      28
    ],
    backdrop_path: '/8BPZO0Bf8TeAy8znF43z8soK3ys.jpg',
    adult: false,
    overview: 'Aragorn is revealed as the heir to the ancient kings as he, Gandalf and the other members of the broken fellowship struggle to save Gondor from Sauron\'s forces. Meanwhile, Frodo and Sam bring the ring closer to the heart of Mordor, the dark lord\'s realm.',
    release_date: '2003-12-01'
  },
  {
    vote_count: 9090,
    id: 120,
    video: false,
    vote_average: 8.1,
    title: 'The Lord of the Rings: The Fellowship of the Ring',
    popularity: 36.635241,
    poster_path: '/bxVxZb5O9OxCO0oRUNdCnpy9NST.jpg',
    original_language: 'en',
    original_title: 'The Lord of the Rings: The Fellowship of the Ring',
    genre_ids: [
      12,
      14,
      28
    ],
    backdrop_path: '/pIUvQ9Ed35wlWhY2oU6OmwEsmzG.jpg',
    adult: false,
    overview: 'Young hobbit Frodo Baggins, after inheriting a mysterious ring from his uncle Bilbo, must leave his home in order to keep it from falling into the hands of its evil creator. Along the way, a fellowship is formed to protect the ringbearer and make sure that the ring arrives at its final destination: Mt. Doom, the only place where it can be destroyed.',
    release_date: '2001-12-18'
  },
  {
    vote_count: 7842,
    id: 121,
    video: false,
    vote_average: 8,
    title: 'The Lord of the Rings: The Two Towers',
    popularity: 24.591994,
    poster_path: '/wf3v0Pn09jnT5HSaYal7Ami3bdA.jpg',
    original_language: 'en',
    original_title: 'The Lord of the Rings: The Two Towers',
    genre_ids: [
      12,
      14,
      28
    ],
    backdrop_path: '/dG4BmM32XJmKiwopLDQmvXEhuHB.jpg',
    adult: false,
    overview: 'Frodo and Sam are trekking to Mordor to destroy the One Ring of Power while Gimli, Legolas and Aragorn search for the orc-captured Merry and Pippin. All along, nefarious wizard Saruman awaits the Fellowship members at the Orthanc Tower in Isengard.',
    release_date: '2002-12-18'
  }
]

export const yodaMovies = [{
    vote_count: 1,
    id: 371796,
    video: false,
    vote_average: 7,
    title: 'The New Yoda Chronicles: Escape from the Jedi Temple',
    popularity: 1.677475,
    poster_path: '/2nwar7SCeY9fktHRA8LFwHRBqiI.jpg',
    original_language: 'en',
    original_title: 'The New Yoda Chronicles: Escape from the Jedi Temple',
    genre_ids: [
      16,
      10751
    ],
    backdrop_path: null,
    adult: false,
    overview: '',
    release_date: '2014-12-09'
  },
  {
    vote_count: 1,
    id: 424492,
    video: false,
    vote_average: 7,
    title: 'The New Yoda Chronicles: Clash of the Skywalkers',
    popularity: 1.053418,
    poster_path: '/d0nFx7eQvXsGZgRkVLY7NymtZXd.jpg',
    original_language: 'en',
    original_title: 'The New Yoda Chronicles: Clash of the Skywalkers',
    genre_ids: [
      16
    ],
    backdrop_path: null,
    adult: false,
    overview: 'The enraged Emperor sends Darth Vader and his entire fleet to find and destroy Luke as revenge for his destruction of the Holocrons. Luke, however, finds inspiration from the last remaining Holocron -- a daring move pulled off years earlier by his father, Anakin. The emboldened Luke uses Anakin\'s daring move to lure Vader and the Imperial Fleet into a trap that could turn the tide for the Rebellion - or lead to disaster.',
    release_date: '2014-11-23'
  }
]

export const movie = {
  adult: false,
  backdrop_path: '/fUjPnCmmBVHLNdgz2sqb1zp39Pr.jpg',
  belongs_to_collection: {
    id: 1241,
    name: 'Harry Potter Collection',
    poster_path: '/eVPs2Y0LyvTLZn6AP5Z6O2rtiGB.jpg',
    backdrop_path: '/wfnMt6LGqYHcNyOfsuusw5lX3bL.jpg'
  },
  budget: 125000000,
  genres: [{
      id: 12,
      name: 'Adventure'
    },
    {
      id: 14,
      name: 'Fantasy'
    },
    {
      id: 10751,
      name: 'Family'
    }
  ],
  homepage: 'http://harrypotter.warnerbros.com/harrypotterandthedeathlyhallows/mainsite/index.html',
  id: 671,
  imdb_id: 'tt0241527',
  original_language: 'en',
  original_title: 'Harry Potter and the Philosopher\'s Stone',
  overview: 'Harry Potter has lived under the stairs at his aunt and uncle\'s house his whole life. But on his 11th birthday, he learns he\'s a powerful wizard -- with a place waiting for him at the Hogwarts School of Witchcraft and Wizardry. As he learns to harness his newfound powers with the help of the school\'s kindly headmaster, Harry uncovers the truth about his parents\' deaths -- and about the villain who\'s to blame.',
  popularity: 32.913022,
  poster_path: '/dCtFvscYcXQKTNvyyaQr2g2UacJ.jpg',
  production_companies: [{
      name: '1492 Pictures',
      id: 436
    },
    {
      name: 'Warner Bros.',
      id: 6194
    },
    {
      name: 'Heyday Films',
      id: 7364
    }
  ],
  production_countries: [{
      iso_3166_1: 'GB',
      name: 'United Kingdom'
    },
    {
      iso_3166_1: 'US',
      name: 'United States of America'
    }
  ],
  release_date: '2001-11-16',
  revenue: 976475550,
  runtime: 152,
  spoken_languages: [{
    iso_639_1: 'en',
    name: 'English'
  }],
  status: 'Released',
  tagline: 'Let the Magic Begin.',
  title: 'Harry Potter and the Philosopher\'s Stone',
  video: false,
  vote_average: 7.5,
  vote_count: 7441
};

export const anotherMovie = {
  vote_count: 9090,
  id: 120,
  video: false,
  vote_average: 8.1,
  title: 'The Lord of the Rings: The Fellowship of the Ring',
  popularity: 36.635241,
  poster_path: '/bxVxZb5O9OxCO0oRUNdCnpy9NST.jpg',
  original_language: 'en',
  original_title: 'The Lord of the Rings: The Fellowship of the Ring',
  genre_ids: [
    12,
    14,
    28
  ],
  backdrop_path: '/pIUvQ9Ed35wlWhY2oU6OmwEsmzG.jpg',
  adult: false,
  overview: 'Young hobbit Frodo Baggins, after inheriting a mysterious ring from his uncle Bilbo, must leave his home in order to keep it from falling into the hands of its evil creator. Along the way, a fellowship is formed to protect the ringbearer and make sure that the ring arrives at its final destination: Mt. Doom, the only place where it can be destroyed.',
  release_date: '2001-12-18'
}

export const FAKE_ACTION = 'FAKE_ACTION';

export const initialState = {
  movies: [],
  selectedMovie: null,
  sortBy: 'release'
};