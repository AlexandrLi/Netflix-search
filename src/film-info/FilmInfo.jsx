import React from 'react';
import { Link } from 'react-router-dom'
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as s from './film-info.css';
import { FilmCard } from '../film-card/FilmCard';
import { fetchMovie, clearMovies } from '../actions';

class FilmInfo extends React.Component {

  componentWillMount() {
    this.props.fetchMovie(this.props.match.params.id);
  }

  // componentWillReceiveProps(nextProps) {
  //   let movie = this.state.movies.find(movie => movie.show_title === nextProps.match.params.show_title)
  //   this.setState({ selectedMovie: movie })
  // }

  render() {
    if (!this.props.selectedMovie) {
      return <div>Loading...</div>
    } else {
      return (
        <div>
          <div className={s.header}>
            <div className={s.row}>
              <h1>netflixroulette</h1>
              <div className={s.button} onClick={this.props.clearMovies}><Link to="/">search</Link></div>
            </div>
            <div className={s.info}>
              {<img src={`https://image.tmdb.org/t/p/w342${this.props.selectedMovie.poster_path}`} alt="no image" />}
              <div className={s.description}>
                <h2>{this.props.selectedMovie.title}</h2>
                <p>{this.props.selectedMovie.genres.map(g => g.name).join(', ')}</p>
                <p>{this.props.selectedMovie.release_date.slice(0, 4)}<span>{this.props.selectedMovie.runtime}</span></p>
                <p>{this.props.selectedMovie.overview}</p>
                <p>Budget: {this.props.selectedMovie.budget}</p>
                <p>Cast: {this.props.selectedMovie.show_cast}</p>
              </div>
            </div>
          </div>
          {!!this.props.selectedMovie.belongs_to_collection &&
            <div className={s.toolbar}>
              <p>Belongs to: {this.props.selectedMovie.belongs_to_collection.name}</p>
            </div>
          }
          <div className={s.container}>
            {/* {this.props.movies.filter(movie => movie.show_id !== this.state.selectedMovie.show_id).map(item => <Link to={`/film/${item.show_title}`} key={item.show_id} ><FilmCard key={item.show_id} movie={item} /></Link>)} */}
          </div>
        </div >
      )
    }
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ fetchMovie, clearMovies }, dispatch);
}

function mapStateToProps({ movies, selectedMovie }) {
  return { movies, selectedMovie };
}

export default connect(mapStateToProps, mapDispatchToProps)(FilmInfo);