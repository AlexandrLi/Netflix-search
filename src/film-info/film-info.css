.header {
  padding: 0 100px 50px 100px;
  color: white;
  position: relative;
  background: rgba(0, 0, 0, 0.75);
  overflow: hidden;
}

.header:before {
  content: "";
  height: 100%;
  width: 100%;
  position: absolute;
  z-index: -1;
  background: url('https://assets.nflxext.com/ffe/siteui/vlv3/dc18d1c3-9e06-41a4-a5b0-409388e3f9e8/532f465f-9176-4555-8e82-11ec6e34201e/KZ-en-20170911-popsignuptwoweeks-perspective_alpha_website_large.jpg');
  transition: transform 1s;
  left: 0;
  top: 0;
}

.header:hover:before {
  transform: scale(1.1);
}

h1 {
  color: #F55263;
  margin: 15px 0;
  font-size: 18px;
}

h2 {
  color: #F55263;
  font-size: 42px;
}

.container {
  display: flex;
  flex-wrap: wrap;
}

.info {
  display: flex;
  align-items: center;
  justify-content: space-around;
  align-items: flex-start;
}

.row {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.description {
  width: 600px;
}

.description p {
  margin: 25px 0;
  font-size: 18px;
}

.description span {
  margin-left: 20px;
}

.button {
  cursor: pointer;
  margin-left: 20px;
  min-width: 50px;
  padding: 5px 10px;
  text-align: center;
  color: #F55263;
  background-color: white;
  border-radius: 2px;
  text-transform: uppercase;
}

.toolbar {
  background-color: #F5F5F5;
  padding: 20px 0;
}

.toolbar p {
  margin-left: 100px;
  font-size: inherit;
}