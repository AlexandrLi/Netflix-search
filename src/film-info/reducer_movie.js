export default function (state = null, action) {
  console.log(action.payload);
  switch (action.type) {
    case 'SELECT_MOVIE':
      return action.payload;
    default:
      return state;
  }
}