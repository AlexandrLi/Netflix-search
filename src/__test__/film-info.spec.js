import React from "react"
import renderer from "react-test-renderer"
import { MemoryRouter } from 'react-router-dom'
import { shallow } from "enzyme";
import { FilmInfo } from "../film-info/FilmInfo";
import { movies, movie } from './mock-data';

const props = {
  movies: movies,
  selectedMovie: movie,
  match: {
    params: {
      id: movie.id
    }
  },
  fetchMovie: jest.fn()
}

describe("Film info snapshot", () => {
  it("Take film-info snapshot", () => {
    const renderedSearchPage = renderer.create(<MemoryRouter><FilmInfo {...props} /></MemoryRouter>).toJSON();
    expect(renderedSearchPage).toMatchSnapshot();
  })
});

describe("Shalow render of Film info component", () => {
  let component;
  beforeEach(() => {
    component = shallow(<MemoryRouter><FilmInfo {...props} /></MemoryRouter>)
  });

  it("Component successfully rendered", () => {
    expect(component.length).toEqual(1);
  });
});