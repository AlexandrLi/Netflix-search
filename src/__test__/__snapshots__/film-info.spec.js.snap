// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film info snapshot Take film-info snapshot 1`] = `
<div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <h1>
        netflixroulette
      </h1>
      <div
        className={undefined}
        onClick={undefined}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          search
        </a>
      </div>
    </div>
    <div
      className={undefined}
    >
      <img
        alt="no image"
        src="https://image.tmdb.org/t/p/w342/dCtFvscYcXQKTNvyyaQr2g2UacJ.jpg"
      />
      <div
        className={undefined}
      >
        <h2>
          Harry Potter and the Philosopher's Stone
        </h2>
        <p>
          Adventure, Fantasy, Family
        </p>
        <p>
          Release year: 
          2001
          <span>
            Runtime: 
            152
          </span>
        </p>
        <p>
          <em>
            Let the Magic Begin.
          </em>
        </p>
        <p>
          Harry Potter has lived under the stairs at his aunt and uncle's house his whole life. But on his 11th birthday, he learns he's a powerful wizard -- with a place waiting for him at the Hogwarts School of Witchcraft and Wizardry. As he learns to harness his newfound powers with the help of the school's kindly headmaster, Harry uncovers the truth about his parents' deaths -- and about the villain who's to blame.
        </p>
        <p>
          Budget: 
          125000000
          <span>
            Rating: 
            7.5
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <p>
      Collection: 
      Harry Potter Collection
    </p>
  </div>
  <div
    className={undefined}
  >
    <a
      href="/film/122"
      onClick={[Function]}
    >
      <div
        className="card"
      >
        <img
          alt="no image"
          src="https://image.tmdb.org/t/p/w342/uexxR7Kw1qYbZk0RYaF9Rx5ykbj.jpg"
        />
        <div
          className="row"
        >
          <h2>
            The Lord of the Rings: The Return of the King
          </h2>
          <div>
            <div
              className="year"
            >
              2003
            </div>
          </div>
        </div>
        <p>
          8.2
        </p>
      </div>
    </a>
    <a
      href="/film/120"
      onClick={[Function]}
    >
      <div
        className="card"
      >
        <img
          alt="no image"
          src="https://image.tmdb.org/t/p/w342/bxVxZb5O9OxCO0oRUNdCnpy9NST.jpg"
        />
        <div
          className="row"
        >
          <h2>
            The Lord of the Rings: The Fellowship of the Ring
          </h2>
          <div>
            <div
              className="year"
            >
              2001
            </div>
          </div>
        </div>
        <p>
          8.1
        </p>
      </div>
    </a>
    <a
      href="/film/121"
      onClick={[Function]}
    >
      <div
        className="card"
      >
        <img
          alt="no image"
          src="https://image.tmdb.org/t/p/w342/wf3v0Pn09jnT5HSaYal7Ami3bdA.jpg"
        />
        <div
          className="row"
        >
          <h2>
            The Lord of the Rings: The Two Towers
          </h2>
          <div>
            <div
              className="year"
            >
              2002
            </div>
          </div>
        </div>
        <p>
          8
        </p>
      </div>
    </a>
  </div>
</div>
`;
